com/ticket.cfc

```diff
@@ -0,0 +1,766 @@
+
+        // Check internal usage 
+        if(structKeyExists(arguments.formStruct, "create_worker")){
+            local.worker = true;
+        }
+
+        // Validate form values 
+        if(structKeyExists(arguments.formStruct, "reference") and len(trim(arguments.formStruct.reference))){
+            session.reference = trim(left(arguments.formStruct.reference, 255));
+        } else {
+            local.argsReturnValue.message = local.worker ? "Enter a reference!" : "#getTrans('txtReferenceError')#";
+        }
+
+        if(structKeyExists(arguments.formStruct, "description") and len(trim(arguments.formStruct.description))){
+            session.description = trim(left(arguments.formStruct.description, 2000));
+        } else {
+
+            if(len(trim(local.argsReturnValue.message))){
+                local.argsReturnValue.message = local.worker ? "Enter a description and reference!" : "#getTrans('txtDescriptionReferenceError')#";
+            } else {
+                local.argsReturnValue.message = local.worker ? "Enter a description!" : "#getTrans('txtDescriptionError')#";
+            }
+
+        }
+
+        // Search for user by email 
+        if(structKeyExists(arguments.formStruct, "create_worker")){
+            if(structKeyExists(arguments.formStruct, "email") and len(trim(arguments.formStruct.email))){
+                
+                // Get user query 
+                try {
+                    local.qUser = queryExecute(
+                        options = {datasource = application.datasource},
+                        params = {
+                            email: {type: "varchar", value: arguments.formStruct.email}
+                        },
+                        sql = "
+                            SELECT intUserID
+                            FROM users
+                            WHERE strEmail = :email
+                        "
+                    )
+                } catch (any e) {
+                    logWrite("Ticketsystem", 3, "File: #callStackGet("string", 0 , 1)#, Could not execute query to find user: #e.message#", false);
+                    local.argsReturnValue.message = "Could not execute query to find user!";
+                    return local.argsReturnValue;
+                }
+
+                // Check if user was found 
+                if(local.qUser.recordCount eq 1){
+                    session.email = arguments.formStruct.email;
+                    local.userID = local.qUser.intUserID;
+                } else {
+                    local.argsReturnValue.message = "E-mail doesn't exists!";
+                    return local.argsReturnValue;
+                }
+
+            } else {
+                local.argsReturnValue.message = "Enter a E-Mail!";
+                return local.argsReturnValue;
+            }
+        }
+
+        // Get ID from user 
+        if(structKeyExists(arguments.formStruct, "create_user")){
+            local.userID = session.user_id;
```

Comment:
You should never work with sessions in a function (cfc). Instead use the handler to send sessions (IDs) into the function.